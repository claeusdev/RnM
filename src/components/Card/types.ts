import { Character } from "types";

export interface CharacterProps {
    character: Character;
  }