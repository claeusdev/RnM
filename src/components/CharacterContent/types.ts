import { Character, Location } from "types";


export interface CharacterContentProps {
  gender: string;
  location: Location;
  origin: { name: string };
  species: string;
}
