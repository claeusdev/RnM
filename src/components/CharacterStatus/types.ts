export interface CharacterStatusProps {
  status: string;
}
