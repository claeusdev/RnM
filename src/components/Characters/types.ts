import { Character } from "types";

export type CharactersProps = {
    page: string
    results: Character[]
}