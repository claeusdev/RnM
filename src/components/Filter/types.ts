export interface FilterProps {
    updatePageNumber: (pageNumber: number) => void;
}
