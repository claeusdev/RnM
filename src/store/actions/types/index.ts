export enum ActionTypes {
    LOAD_CHARACTERS = "LOAD_CHARACTERS",
    LOAD_CHARACTERS_SUCCESS = "LOAD_CHARACTERS_SUCCESS",
    LOAD_CHARACTERS_ERROR = "LOAD_CHARACTERS_ERROR",
    FILTER_CHARACTERS = "FILTER_CHARACTERS",
    FILTER_CHARACTERS_SUCCESS = "FILTER_CHARACTERS_SUCCESS",
    FILTER_CHARACTERS_ERROR = "FILTER_CHARACTERS_ERROR",
    LOAD_CHARACTER = "LOAD_CHARACTER",
    LOAD_CHARACTER_SUCCESS = "LOAD_CHARACTER_SUCCESS",
    LOAD_CHARACTER_ERROR = "LOAD_CHARACTER_ERROR",
    SET_EPISODES = "SET_EPISODES"
}