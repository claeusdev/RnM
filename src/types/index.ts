export interface State {
  characters: Character[],
  loading: boolean;
  error: string | null;
}
export interface Action {
  type: string;
  payload?: CharacterResponse | string;
}
export interface Character {
  id: number;
  name: string;
  species: string;
  image: string;
  status: string;
  origin: Location;
  location: Location;
  gender: string;
  episode: string[];
  url: string;
}

export type GetCharacterParams = {
  status?: string;
  species?: string;
  pageNumber?: number,
  gender?: string;
}
export interface Location {
  name: string;
  url: string;
}

export interface CharacterResponse {
  info: Info;
  results: Character[];
}

export type Info = {
  count: number;
  pages: number;
  next: string;
  prev: string | null;
};


export type Episode = {
  id: number;
  name: string;
  air_date: string;
  episode: string;
  characters: Character[],
  url: string;
  created: string;
}