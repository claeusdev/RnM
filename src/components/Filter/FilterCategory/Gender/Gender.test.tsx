import { render } from "@testing-library/react";
import { testCharacter } from "types/testData";
import GenderFilter from "./Gender";

test("renders Gender Filter", () => {
  const component = render(<GenderFilter updateGender={jest.fn()} updatePageNumber={jest.fn()}/>);
  expect(component).toMatchSnapshot();
});
