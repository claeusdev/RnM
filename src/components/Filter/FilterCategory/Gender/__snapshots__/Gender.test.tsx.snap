// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Gender Filter 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="accordion-item"
      >
        <h2
          class="accordion-header"
          id="headingThree"
        >
          <button
            aria-controls="collapseThree"
            aria-expanded="false"
            class="accordion-button collapsed"
            data-bs-target="#collapseThree"
            data-bs-toggle="collapse"
            type="button"
          >
            Gender
          </button>
        </h2>
        <div
          aria-labelledby="headingThree"
          class="accordion-collapse collapse"
          data-bs-parent="#accordionExample"
          id="collapseThree"
        >
          <div
            class="accordion-body d-flex flex-wrap gap-3"
          >
            <style>
              
          .x:checked + label {
            background-color: #0b5ed7;
            color: white;
          }
          input[type="radio"] {
            display: none;
          }
        
            </style>
            <div
              class="form-check"
            >
              <input
                class="form-check-input x"
                id="gender-0"
                name="gender"
                type="radio"
              />
              <label
                class="filterBtn"
                for="gender-0"
              >
                female
              </label>
            </div>
            <style>
              
          .x:checked + label {
            background-color: #0b5ed7;
            color: white;
          }
          input[type="radio"] {
            display: none;
          }
        
            </style>
            <div
              class="form-check"
            >
              <input
                class="form-check-input x"
                id="gender-1"
                name="gender"
                type="radio"
              />
              <label
                class="filterBtn"
                for="gender-1"
              >
                male
              </label>
            </div>
            <style>
              
          .x:checked + label {
            background-color: #0b5ed7;
            color: white;
          }
          input[type="radio"] {
            display: none;
          }
        
            </style>
            <div
              class="form-check"
            >
              <input
                class="form-check-input x"
                id="gender-2"
                name="gender"
                type="radio"
              />
              <label
                class="filterBtn"
                for="gender-2"
              >
                genderless
              </label>
            </div>
            <style>
              
          .x:checked + label {
            background-color: #0b5ed7;
            color: white;
          }
          input[type="radio"] {
            display: none;
          }
        
            </style>
            <div
              class="form-check"
            >
              <input
                class="form-check-input x"
                id="gender-3"
                name="gender"
                type="radio"
              />
              <label
                class="filterBtn"
                for="gender-3"
              >
                unknown
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="accordion-item"
    >
      <h2
        class="accordion-header"
        id="headingThree"
      >
        <button
          aria-controls="collapseThree"
          aria-expanded="false"
          class="accordion-button collapsed"
          data-bs-target="#collapseThree"
          data-bs-toggle="collapse"
          type="button"
        >
          Gender
        </button>
      </h2>
      <div
        aria-labelledby="headingThree"
        class="accordion-collapse collapse"
        data-bs-parent="#accordionExample"
        id="collapseThree"
      >
        <div
          class="accordion-body d-flex flex-wrap gap-3"
        >
          <style>
            
          .x:checked + label {
            background-color: #0b5ed7;
            color: white;
          }
          input[type="radio"] {
            display: none;
          }
        
          </style>
          <div
            class="form-check"
          >
            <input
              class="form-check-input x"
              id="gender-0"
              name="gender"
              type="radio"
            />
            <label
              class="filterBtn"
              for="gender-0"
            >
              female
            </label>
          </div>
          <style>
            
          .x:checked + label {
            background-color: #0b5ed7;
            color: white;
          }
          input[type="radio"] {
            display: none;
          }
        
          </style>
          <div
            class="form-check"
          >
            <input
              class="form-check-input x"
              id="gender-1"
              name="gender"
              type="radio"
            />
            <label
              class="filterBtn"
              for="gender-1"
            >
              male
            </label>
          </div>
          <style>
            
          .x:checked + label {
            background-color: #0b5ed7;
            color: white;
          }
          input[type="radio"] {
            display: none;
          }
        
          </style>
          <div
            class="form-check"
          >
            <input
              class="form-check-input x"
              id="gender-2"
              name="gender"
              type="radio"
            />
            <label
              class="filterBtn"
              for="gender-2"
            >
              genderless
            </label>
          </div>
          <style>
            
          .x:checked + label {
            background-color: #0b5ed7;
            color: white;
          }
          input[type="radio"] {
            display: none;
          }
        
          </style>
          <div
            class="form-check"
          >
            <input
              class="form-check-input x"
              id="gender-3"
              name="gender"
              type="radio"
            />
            <label
              class="filterBtn"
              for="gender-3"
            >
              unknown
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders character location/status 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="accordion-item"
      >
        <h2
          class="accordion-header"
          id="headingThree"
        >
          <button
            aria-controls="collapseThree"
            aria-expanded="false"
            class="accordion-button collapsed"
            data-bs-target="#collapseThree"
            data-bs-toggle="collapse"
            type="button"
          >
            Gender
          </button>
        </h2>
        <div
          aria-labelledby="headingThree"
          class="accordion-collapse collapse"
          data-bs-parent="#accordionExample"
          id="collapseThree"
        >
          <div
            class="accordion-body d-flex flex-wrap gap-3"
          >
            <style>
              
          .x:checked + label {
            background-color: #0b5ed7;
            color: white;
          }
          input[type="radio"] {
            display: none;
          }
        
            </style>
            <div
              class="form-check"
            >
              <input
                class="form-check-input x"
                id="gender-0"
                name="gender"
                type="radio"
              />
              <label
                class="btn btn-outline-primary"
                for="gender-0"
              >
                female
              </label>
            </div>
            <style>
              
          .x:checked + label {
            background-color: #0b5ed7;
            color: white;
          }
          input[type="radio"] {
            display: none;
          }
        
            </style>
            <div
              class="form-check"
            >
              <input
                class="form-check-input x"
                id="gender-1"
                name="gender"
                type="radio"
              />
              <label
                class="btn btn-outline-primary"
                for="gender-1"
              >
                male
              </label>
            </div>
            <style>
              
          .x:checked + label {
            background-color: #0b5ed7;
            color: white;
          }
          input[type="radio"] {
            display: none;
          }
        
            </style>
            <div
              class="form-check"
            >
              <input
                class="form-check-input x"
                id="gender-2"
                name="gender"
                type="radio"
              />
              <label
                class="btn btn-outline-primary"
                for="gender-2"
              >
                genderless
              </label>
            </div>
            <style>
              
          .x:checked + label {
            background-color: #0b5ed7;
            color: white;
          }
          input[type="radio"] {
            display: none;
          }
        
            </style>
            <div
              class="form-check"
            >
              <input
                class="form-check-input x"
                id="gender-3"
                name="gender"
                type="radio"
              />
              <label
                class="btn btn-outline-primary"
                for="gender-3"
              >
                unknown
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="accordion-item"
    >
      <h2
        class="accordion-header"
        id="headingThree"
      >
        <button
          aria-controls="collapseThree"
          aria-expanded="false"
          class="accordion-button collapsed"
          data-bs-target="#collapseThree"
          data-bs-toggle="collapse"
          type="button"
        >
          Gender
        </button>
      </h2>
      <div
        aria-labelledby="headingThree"
        class="accordion-collapse collapse"
        data-bs-parent="#accordionExample"
        id="collapseThree"
      >
        <div
          class="accordion-body d-flex flex-wrap gap-3"
        >
          <style>
            
          .x:checked + label {
            background-color: #0b5ed7;
            color: white;
          }
          input[type="radio"] {
            display: none;
          }
        
          </style>
          <div
            class="form-check"
          >
            <input
              class="form-check-input x"
              id="gender-0"
              name="gender"
              type="radio"
            />
            <label
              class="btn btn-outline-primary"
              for="gender-0"
            >
              female
            </label>
          </div>
          <style>
            
          .x:checked + label {
            background-color: #0b5ed7;
            color: white;
          }
          input[type="radio"] {
            display: none;
          }
        
          </style>
          <div
            class="form-check"
          >
            <input
              class="form-check-input x"
              id="gender-1"
              name="gender"
              type="radio"
            />
            <label
              class="btn btn-outline-primary"
              for="gender-1"
            >
              male
            </label>
          </div>
          <style>
            
          .x:checked + label {
            background-color: #0b5ed7;
            color: white;
          }
          input[type="radio"] {
            display: none;
          }
        
          </style>
          <div
            class="form-check"
          >
            <input
              class="form-check-input x"
              id="gender-2"
              name="gender"
              type="radio"
            />
            <label
              class="btn btn-outline-primary"
              for="gender-2"
            >
              genderless
            </label>
          </div>
          <style>
            
          .x:checked + label {
            background-color: #0b5ed7;
            color: white;
          }
          input[type="radio"] {
            display: none;
          }
        
          </style>
          <div
            class="form-check"
          >
            <input
              class="form-check-input x"
              id="gender-3"
              name="gender"
              type="radio"
            />
            <label
              class="btn btn-outline-primary"
              for="gender-3"
            >
              unknown
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
